GarNet/garnet.py

```diff
@@ -324,221 +242,54 @@ def TF_regression(motifs_and_genes_file_or_dataframe, expression_file, output_di
+	# reorder columns
+	motif_genes_df = motif_genes_df[["motifChrom", "motifStart", "motifEnd", "motifName", "motifScore", "motifStrand",
+									 "geneName", "tssStart", "tssEnd", "motif_gene_distance"]]
 
-	else:
-		starts = reference.apply(lambda x: x.geneStart - upstream_window, axis=1)
-		ends = reference.apply(lambda x: x.geneEnd + downstream_window, axis=1)
-
-	intervals = zip(starts.values, ends.values, reference.to_dict(orient='records'))
-
-	tree = IntervalTree.from_tuples(intervals)
-
-	return tree
-
-
-def IntervalTree_from_motifs(motifs):
-	"""
-	Arguments:
-		motifs (dataframe): Must be a dataframe with motifStart and motifEnd columns
-
-	Returns:
-		IntervalTree: of motifs
-	"""
-
-	motifs = motifs.dropna(subset=['motifStart', 'motifEnd'])
-
-	intervals = zip(motifs.motifStart.astype(int).values, motifs.motifEnd.astype(int).values, motifs.to_dict(orient='records'))
-
-	tree = IntervalTree.from_tuples(intervals)
-
-	return tree
-
-
-def intersection_of_dict_of_intervaltree(A, B):
-	"""
-	Arguments:
-		A (dict): is a dictionary of {chrom (str): IntervalTree}
-		B (dict): is a dictionary of {chrom (str): IntervalTree}
-
-	Returns:
-		dict: {keys shared between A and B: {intervals in A: [list of overlapping intervals in B]} }
-	"""
-
-	logger.info('Computing intersection operation of IntervalTrees for each chromosome...')
-
-	# Keys are chromosomes. We only want to look through chromosomes where there is potential for overlap
-	common_keys = set(A.keys()).intersection( set(B.keys()) )
-
-	# In general, the below operation isn't perfectly elegant, due to the double-for:
-	#   `for key in common_keys for a in A[key]`
-	# The reason we use a double-for here is because we want to do one level of "flattening"
-	# but we don't want to do it as a pre-processing or post-processing step. Specifically,
-	# we're passed dictionaries of chrom: IntervalTree, and we'd like to return a single data
-	# structure for the entire genome, not one per chromosome. The below can be read as:
-	#
-	#	for chromosome in chromosomes_found_in_both_datastructures:
-	#		for a_interval in A[chromosome]
-	#			for b_interval in B[chromosome].search(A_interval)
-	#				Map a_interval to b_interval
-	#
-	# which can be expressed concisely in the following line of code (which is also maximally efficient)
-	intersection = [(a.data, b.data) for key in common_keys for a in A[key] for b in B[key].search(a)]
-
-	return intersection
-
-
-def type_of_peak(row):
-	"""
-	Arguments:
-		row (pd.Series): A row of data from a dataframe with peak and gene information
-
-	Returns:
-		str: a name for the relationship between the peak and the gene:
-				- upstream if the start of the peak is more than 2kb above the start of the gene
-				- promoter if the start of the peak is above the start of the gene
-				- downstream if the start of the peak is below the start of the gene
-	"""
+	motif_genes_df.to_csv(output_file, sep='\t', index=False, header=False)
```

Comment:
why `header=False`?

Reply:
Easier for bedtools to handle...on second pass, it looks like bedtools may have functionality to handle headers, will add this in a future commit.